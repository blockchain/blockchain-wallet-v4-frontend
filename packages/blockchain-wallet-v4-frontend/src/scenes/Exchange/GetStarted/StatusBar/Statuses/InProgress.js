import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'

import { actions } from 'data'
import { Button } from 'blockchain-info-components'

const ActionButton = styled(Button).attrs({ nature: 'primary' })`
  font-weight: 500;
`

export const InProgress = ({ verifyIdentity }) => (
  <ActionButton onClick={verifyIdentity}>
    <FormattedMessage
      id='scenes.exchange.getstarted.status.inprogress.button'
      defaultMessage='Continue where you left off'
    />
  </ActionButton>
)

const mapDispatchToProps = dispatch => ({
  verifyIdentity: () =>
    dispatch(actions.components.identityVerification.verifyIdentity())
})

export default connect(
  undefined,
  mapDispatchToProps
)(InProgress)
