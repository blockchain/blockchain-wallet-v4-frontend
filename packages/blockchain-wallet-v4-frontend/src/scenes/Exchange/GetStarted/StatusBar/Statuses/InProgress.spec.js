import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { InProgress } from './InProgress'

jest.mock('blockchain-info-components', () => ({
  Button: 'button'
}))

describe('InProgress', () => {
  it('renders correctly', () => {
    const component = shallow(<InProgress />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
