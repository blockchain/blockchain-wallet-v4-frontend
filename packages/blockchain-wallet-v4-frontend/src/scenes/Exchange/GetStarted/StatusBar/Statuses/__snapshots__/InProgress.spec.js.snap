// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InProgress renders correctly 1`] = `
<InProgress__ActionButton>
  <FormattedMessage
    defaultMessage="Continue where you left off"
    id="scenes.exchange.getstarted.status.inprogress.button"
    values={Object {}}
  />
</InProgress__ActionButton>
`;
